nixos/modules/services/editors/vim.nix

```diff
@@ -0,0 +1,33 @@
+{ config, lib, pkgs, ...}:
+
+with lib;
+
+let
+
+  cfg = config.services.vim;
+
+in {
+
+  options.services.vim =  {
+    defaultEditor = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Set VIM to be the default editor";
+    };
+  };
+
+    config = mkIf cfg.defaultEditor {
+      environment.systemPackages = [ pkgs.vimnox ];
```

Comment:
I was talking about a configurable [package option](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/editors/emacs.nix#L48) used by the rest of the module, `pkgs.vim` is fine as a default.